### ISSUE TITLE
Opam forget installed root when reinstalling

### ISSUE THREAD

**Issue by Julow** [ocaml/opam#3430]
An example with `cohttp-lwt-unix`:

- I set it as root: `opam install --set-root cohttp-lwt-unix`
- I *remove* a dependency's *optional* dependency: `opam remove js_of_ocaml`
- `cohttp-lwt-unix` is rebuild and is no longer a root

The bug does not occur with `install` and `reinstall`. Only `remove`

**Comment by AltGr** [comment]
Thanks for reporting.
Would you mind specifying at least the opam version concerned ? And preferably the output of `opam config report` ?

**Comment by Julow** [comment]
Hi,

```
# OPAM config report
# opam-version    1.2.2 
# self-upgrade    no
# os              linux
# external-solver aspcud $in $out $criteria
# criteria        -count(removed),-notuptodate(request),-sum(request,version-lag),-count(down),-notuptodate(changed),-count(changed),-notuptodate(solution),-sum(solution,version-lag)
# jobs            4
# repositories    1* (http)
# pinned          0
# current-switch  4.06.1
# last-update     2018-06-21 09:56
```

**Comment by rjbou** [comment]
It is a problem specific to opam1.2, it is fixed in 2.0.
Thanks for reporting!
